source/CINetworkMessageManager.cpp

```diff
@@ -111,15 +112,15 @@ void NetworkMessageManager::receiveMessages() {
 
     _conn->receive([this](const std::vector<uint8_t>& recv) {
         if (!recv.empty()) {
-            uint8_t message_type = recv[0];
+            int message_type = NetworkUtils::decodeInt(recv[0], recv[1], recv[2], recv[3]);
```

Comment:
did we change the size of `message_type`?

Reply:
Yeah I made it so message type now gets marshalled using encode and decode int